x-pack/platform/plugins/shared/lens/public/react_embeddable/helper.ts

```diff
@@ -172,3 +180,16 @@ export function extractInheritedViewModeObservable(
   }
   return new BehaviorSubject<ViewMode>('view');
 }
+
+export function getStructuredDatasourceStates(
+  datasourceStates?: GeneralDatasourceStates
+): StructuredDatasourceStates {
+  return {
+    formBased: ((datasourceStates as DatasourceStates)?.formBased?.state ??
+      datasourceStates?.formBased ??
```

Comment:
nit: why falling back to `undefined` ?

Reply:
I thought to just fully align the type as `formBased?: FormBasedPersistedState` and avoid the possibility of `formBased?: FormBasedPersistedState | null`.